File: 151002/Rusakovich/NewsDistributedSystem/discussion/src/main/java/by/rusakovich/discussion/model/Note.java
package by.rusakovich.discussion.model;


import com.datastax.oss.driver.api.mapper.annotations.NamingStrategy;
import lombok.*;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;

@Getter
@Setter
@NamingStrategy(customConverterClass = DCNameConvention.class)
public class Note {
    @PrimaryKeyColumn(name = "country", type= PrimaryKeyType.PARTITIONED)
    private String country;
    @PrimaryKeyColumn(name = "news_id", ordinal = 0, type = PrimaryKeyType.CLUSTERED)
    private Long newsId;
    @PrimaryKeyColumn(name = "id", ordinal = 1, type= PrimaryKeyType.CLUSTERED)
    private Long id;
    private String content;
}

File: 151004/Bosko/lab3/lab3/Discussion/NoteEntity/Note.cs
using Discussion.Common;

namespace Discussion.NoteEntity
{
    public class Note(string content, string country) : AbstractEntity
    {
        public string Country { get; set; } = country;
        public int NewsId { get; set; }
        public string Content { get; set; } = content;

        public Note() : this(string.Empty, string.Empty) { }

        public Note(int id, int newsId, string content) : this(content, string.Empty)
        {
            Id = id;
            NewsId = newsId;
        }

        public Note(int id, int newsId, string content, string country) : this(content, country)
        {
            Id = id;
            NewsId = newsId;
        }
    }
}
